File: dotnet/test/Azure.Iot.Operations.Protocol.MetlTests/TestCasePublishedMessage.cs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

namespace Azure.Iot.Operations.Protocol.MetlTests
{
    public class TestCasePublishedMessage
    {
        public int? CorrelationIndex { get; set; }

        public string? Topic { get; set; }

        public object Payload { get; set; } = false;

        public string? ContentType { get; set; }

        public int? FormatIndicator { get; set; }

        public Dictionary<string, string?> Metadata { get; set; } = new();

        public object CommandStatus { get; set; } = false;

        public bool? IsApplicationError { get; set; }

        public string? SourceId { get; set; }

        public int? Expiry { get; set; }
    }
}

File: rust/azure_iot_operations_protocol/tests/metl/test_case_published_message.rs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

use std::collections::HashMap;

use serde::Deserialize;

use crate::metl::optional_field::deserialize_optional_field;

#[derive(Clone, Deserialize, Debug)]
#[allow(dead_code)]
pub struct TestCasePublishedMessage {
    #[serde(rename = "correlation-index")]
    pub correlation_index: Option<i32>,

    #[serde(rename = "topic")]
    pub topic: Option<String>,

    #[serde(rename = "payload")]
    #[serde(default)]
    #[serde(deserialize_with = "deserialize_optional_field")]
    #[allow(clippy::option_option)]
    pub payload: Option<Option<String>>,

    #[serde(rename = "metadata")]
    #[serde(default)]
    pub metadata: HashMap<String, Option<String>>,

    #[serde(rename = "command-status")]
    #[serde(default)]
    #[serde(deserialize_with = "deserialize_optional_field")]
    #[allow(clippy::option_option)]
    pub command_status: Option<Option<i32>>,

    #[serde(rename = "is-application-error")]
    pub is_application_error: Option<bool>,

    #[serde(rename = "source-id")]
    pub source_id: Option<String>,

    #[serde(rename = "expiry")]
    pub expiry: Option<u32>,
}
